controllers/cloud.redhat.com/insightsapp_controller.go

```diff
@@ -73,13 +73,22 @@ func (r *InsightsAppReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, err
 	}
 
-	maker := makers.Maker{
+	maker, err := makers.New(&makers.Maker{
```

Comment:
Inside of `New` the intention is to look at the Base config and instantiate the correct objects based on the `Provider` values.  In the future we'd remove `App: &iapp` from the constructor.